core/src/main/java/com/linecorp/armeria/client/endpoint/DynamicEndpointGroup.java

```diff
@@ -52,8 +51,9 @@
     private final AtomicReference<EndpointSelector> selector = new AtomicReference<>();
     private volatile List<Endpoint> endpoints = UNINITIALIZED_ENDPOINTS;
     private final Lock endpointsLock = new ReentrantLock();
-    private final CompletableFuture<List<Endpoint>> initialEndpointsFuture =
-            new EventLoopCheckingFuture<>();
+
+    private final CompletableFuture<Void> initialEndpointsSet = new CompletableFuture<>();
```

Comment:
Could you add a comment about why we're not using `EventLoopCheckingFuture` here?

Reply:
Changed to use `EventLoopCheckingFuture` for all dynamic endpoint group. 😄